lighthouse-core/audits/manifest-short-name-length.js

```diff
@@ -28,26 +27,24 @@ class ManifestShortNameLength extends Audit {
    * @param {!Artifacts} artifacts
    * @return {!AuditResult}
    */
-  static audit(artifacts) {
+  static audit_(artifacts) {
+    const failures = [];
+    const warnings = [];
+
     return artifacts.requestManifestValues(artifacts.Manifest).then(manifestValues => {
+      const result = {warnings, failures, manifestValues};
+
       if (manifestValues.isParseFailure) {
-        return {
-          rawValue: false,
-        };
+        failures.push(manifestValues.parseFailureReason);
+        return result;
       }
 
-      const hasShortName = manifestValues.allChecks.find(i => i.id === 'hasShortName').passing;
-      if (!hasShortName) {
-        return {
-          rawValue: false,
-          debugString: 'No short_name found in manifest.',
-        };
-      }
+      const shortNameCheckIds = ['hasShortName', 'shortNameLength'];
+      manifestValues.allChecks.filter(item => shortNameCheckIds.includes(item.id)).forEach(item => {
+        if (!item.passing) failures.push(item.failureText);
```

Comment:
won't this have the other problem where if you don't have a short name you'll get both ``'Manifest does not have `short_name`'`` and ``'Manifest `short_name` will be truncated when displayed on the homescreen'``